src/main/scala/sparkz/core/network/peer/PeerManager.scala

```diff
@@ -189,34 +191,26 @@ object PeerManager {
                           sparkzContext: SparkzContext): Map[InetSocketAddress, PeerInfo] = peers.map(p => p._1 -> p._2.peerInfo)
     }
 
-    case class RandomPeerForConnectionExcluding(excludedPeers: Seq[Option[InetSocketAddress]]) extends GetPeers[Option[PeerInfo]] {
+    case class RandomPeerForConnectionExcluding(excludedPeers: Seq[Option[InetSocketAddress]], onlyKnownPeers: Boolean = false) extends GetPeers[Option[PeerInfo]] {
       private val secureRandom = new SecureRandom()
 
       override def choose(peers: Map[InetSocketAddress, PeerDatabaseValue],
                           blacklistedPeers: Seq[InetAddress],
                           sparkzContext: SparkzContext): Option[PeerInfo] = {
-        var response: Option[PeerInfo] = None
-
-        val forgerPeers = peers.filter(_._2.confidence == PeerConfidence.Forger)
-        val forgerCandidates = forgerPeers.values.filterNot(goodCandidateFilter(excludedPeers, blacklistedPeers, _)).toSeq
-
-        if (forgerCandidates.nonEmpty) {
-          response = Some(forgerCandidates(secureRandom.nextInt(forgerCandidates.size)).peerInfo)
-        } else {
-          val highConfidencePeers = peers.filter(_._2.confidence == PeerConfidence.High)
-          val highConfidenceCandidates = highConfidencePeers.values.filterNot(goodCandidateFilter(excludedPeers, blacklistedPeers, _)).toSeq
-
-          if (highConfidenceCandidates.nonEmpty) {
-            response = Some(highConfidenceCandidates(secureRandom.nextInt(highConfidenceCandidates.size)).peerInfo)
-          } else {
-            val candidates = peers.values.filterNot(goodCandidateFilter(excludedPeers, blacklistedPeers, _)).toSeq
-
-            if (candidates.nonEmpty)
-              response = Some(candidates(secureRandom.nextInt(candidates.size)).peerInfo)
-          }
-        }
-
-        response
+        val candidates: Map[PeerConfidence, Seq[PeerDatabaseValue]] = peers.values.toSeq
+          .filterNot(goodCandidateFilter(excludedPeers, blacklistedPeers, _))
+          .groupBy(_.confidence)
+
+        val forgerCandidates = candidates.getOrElse(PeerConfidence.Forger, Seq())
+        val highConfidenceCandidates = candidates.getOrElse(PeerConfidence.High, Seq())
```

Comment:
can we rename highConfidenceCandidates => knwonPeersCandidates for better readability?